GOVERNANCE.md

```diff
@@ -22,30 +18,226 @@ New maintainers must be nominated by an existing maintainer and must be elected
+
+#### Performance Considerations
+Any concerns with scalability, performance, or reliability for the
+implementation?
+
+#### Observability Considerations
+Any new log statements or other considerations to make this feature
+observable and debuggable for admin users?
+
+#### Security Considerations
+How does the proposal consider security? What makes the new features
+secure?
+
+#### Usability Considerations
+How does the proposal consider usability for the end user (kubectl user)
+and for the admin user who installs and configures Pinniped?
+
+#### Documentation Considerations
+How will users discover the new features? Will docs changes be required
+during implementation?
+
+### Other Approaches Considered
+Mention of other reasonable ways that the problem(s)
+could be addressed with rationale for why they were less
+desirable than the proposed approach.
+
+## Open Questions
+A list of questions that need to be answered.
+
+## Answered Questions
+A list of questions that have been answered.
+
+## Implementation Plan
+Who will implement this proposal once it is finished and approved?
+Do you already have ideas for how you might approach the implementation
+in an iterative fashion? For a large proposal with an iterative plan,
+where might you draw the line to define a minimum viable version?
+
+## Implementation PRs
+This section is a placeholder to list the PRs that implement this proposal.
+This section should be left empty until after the proposal is approved.
+After implementation, the proposal can be updated to list related
+implementation PRs.
+```
+
+## Proposal States
+| Status | Definition |
+| --- | --- |
+| `draft` | The proposal is actively being written by the proposer. Not yet ready for review. |
+| `in-review` | The proposal is being reviewed by the community and the project maintainers. |
+| `accepted` | The proposal has been accepted by the project maintainers. |
+| `rejected` | The proposal has been rejected by the project maintainers. |
+| `implemented` | The proposal was accepted and has since been implemented. |
+
+## Lifecycle of a Proposal
+1. Author adds a proposal by creating a PR in draft mode. (Authors can save their work until ready.)
+2. When the author elaborates the proposal sufficiently to withstand critique they:
+   1. change the status to `in-review` and
+   2. mark the PR as "Ready for Review".
+3. The community critiques the proposal by adding PR reviews in order to mature/converge on the proposal.
+4. When the maintainers reach consensus or supermajority to accept a proposal, they:
+   1. change the status to `accepted`,
+   2. adjust the proposal number in the subdirectory's name if needed,
+   3. record both majority and dissenting opinions,
+   4. merge the PR, thus adding the new proposal to the `main` branch,
+   5. code implementation PRs are submitted separately to implement the solution.
+5. During implementation of an accepted proposal:
```

Comment:
What if this never happens?  At what point does the accepted proposal become stale?  For SIG Auth, we require that accepted KEPs start implementation work on the same or next release in which the KEP is marked as implementable.

Reply:
Any suggestions for what we should document in our process to handle this case @enj ?

Reply:
@cfryanr lets defer this to a later time.  I am probably too defensive due to spending so much time working on Kube.

Reply:
Ok, thanks.